db/db_impl.cc

```diff
@@ -2474,22 +2474,22 @@ Status DBImpl::DeleteFilesInRanges(ColumnFamilyHandle* column_family,
             i, begin_key, end_key, &level_files, -1 /* hint_index */,
             nullptr /* file_index */);
         FileMetaData* level_file;
+        deleted_files.reserve(level_files.size());
+        edit.ReserveDelete(level_files.size());
         for (uint32_t j = 0; j < level_files.size(); j++) {
           level_file = level_files[j];
           if (level_file->being_compacted) {
             continue;
           }
-          if (deleted_files.find(level_file) != deleted_files.end()) {
```

Comment:
Multiple ranges can overlap and delete the same file. We should dedup at the end. Or maybe use `unordered_set`?

Reply:
Oh, it's due to multiple ranges. I will add it back